platform/darwin/src/MGLEvent.mm

```diff
@@ -0,0 +1,107 @@
+    return value.match(
+                       [](const mapbox::base::NullValue) -> id { return [NSNull null]; },
+                       [](const std::string &value)            { return @(value.c_str()); },
+                       [](bool value)                          { return @(value); },
+                       [](uint64_t value)                      { return @(value); },
+                       [](int64_t value)                       { return @(value); },
+                       [](double value)                        { return @(value); },
+                       [](const mapbox::base::ValueArray& valueArray) {
+                            NSMutableArray *array = [NSMutableArray arrayWithCapacity:valueArray.size()];
+                            for (const auto& v : valueArray) {
+                                [array addObject:MGLJSONObjectFromMapboxBaseValue(v)];
+                            }
+                            return array;
+                       },
+                       [](const mapbox::base::ValueObject& object) {
+                            NSMutableDictionary *dictionary = [NSMutableDictionary dictionaryWithCapacity:object.size()];
+                            for (const auto& kv : object) {
+                                dictionary[@(kv.first.c_str())] = MGLJSONObjectFromMapboxBaseValue(kv.second);
+                            }
+                            return dictionary;
+                       }
+                       );
+}
+
+#pragma mark - Event
+
+@implementation MGLEvent
+
+- (instancetype)init {
+    [self doesNotRecognizeSelector:_cmd];
+    return nil;
+}
+
+- (instancetype)initWithEvent:(const mbgl::ObservableEvent&)event {
+    self = [super init];
+
+    if (!self)
+        return nil;
+
+    // mbgl::ObservableEvent timestamps do not use the system_clock, i.e. they
+    // are not relative to the Unix epoch, so we'll need to offset appropriately
+    auto systemClockNow = std::chrono::system_clock::now();
+    auto steadyClockNow = std::chrono::steady_clock::now();
+    auto begin = std::chrono::time_point_cast<std::chrono::system_clock::duration>(systemClockNow +
+                                                                                   (event.begin - steadyClockNow));
+    auto end = std::chrono::time_point_cast<std::chrono::system_clock::duration>(systemClockNow +
+                                                                                 (event.end - steadyClockNow));
+    
+    auto beginTime  = std::chrono::duration<double, std::ratio<1>>(begin.time_since_epoch()).count();
+    auto endTime    = std::chrono::duration<double, std::ratio<1>>(end.time_since_epoch()).count();
+
+    _type   = (MGLEventType)[NSString stringWithUTF8String:event.type.c_str()];
+    _begin  = beginTime;
+    _end    = endTime;
+//    _begin = [NSDate dateWithTimeIntervalSince1970:beginTime];
+//    _end   = [NSDate dateWithTimeIntervalSince1970:endTime];
```

Comment:
can be removed?

Reply:
Will remove in a later PR. It's there as a reminder to see which feels more appropriate. I'm not sure that creating NSDates (which can be expensive) is worth it in this instance.

Reply:
Done.